lib/state/simperium/middleware.ts

```diff
@@ -43,6 +45,36 @@ const fetchRevisions = (store: S.Store, state: S.State) => {
   );
 };
 
+const fetchPreferences = (store: S.Store) => {
+  const objectKey = 'preferences-key';
+  buckets.preferences.get(objectKey, (e, preferences) => {
+    let analyticsEnabled = preferences?.data?.analytics_enabled;
+
+    // Necessary for legacy compatibility with the iOS version
+    analyticsEnabled = Boolean(analyticsEnabled);
+
+    // Create empty preferences object in the bucket if not yet there
+    if (preferences === undefined) {
+      buckets.preferences.update(objectKey, { analytics_enabled: false });
+    }
+
+    // Global to be checked in analytics.tracks.recordEvent()
+    window.analyticsEnabled = analyticsEnabled;
+
+    store.dispatch(actions.settings.setAnalyticsEnabled(analyticsEnabled));
+  });
+};
+
+const setAnalyticsEnabled = (state: S.State) => {
+  const objectKey = 'preferences-key';
+  buckets.preferences.get(objectKey, (e, preferences) => {
+    buckets.preferences.update(objectKey, {
+      ...preferences.data,
+      analytics_enabled: state.settings.analyticsEnabled,
+    });
+  });
+};
```

Comment:
it might be helpful to leave `state` out of here and focus on the semantic of what we're doing specifically. if we were to stick to one Redux action we could even inline the call.

```ts
const { preferences } = buckets;

case 'TOGGLE_ANALYTICS':
	preferences.get( 'preferences-key', ( e, { data } ) => {
		if ( e ) {
			debug('Failed to get previous preferences data');
			break;
		}

		const analytics_enabled = action.doEnable ?? ! state.settings.analyticsEnabled;

		preferences.update( 'preferences-key', { ...data, analytics_enabled } );
	} );
	break;
```